mmdet/models/dense_heads/yolo_head.py

```diff
@@ -215,200 +215,68 @@ def get_bboxes(self,
+        flatten_strides = torch.cat(flatten_strides)
+        flatten_bboxes = self.bbox_coder.decode(flatten_anchors,
+                                                flatten_bbox_preds,
+                                                flatten_strides.unsqueeze(-1))
+
+        if with_nms and (flatten_objectness.size(0) == 0):
             return torch.zeros((0, 5)), torch.zeros((0, ))
 
         if rescale:
-            batch_mlvl_bboxes /= batch_mlvl_bboxes.new_tensor(
+            flatten_bboxes /= flatten_bboxes.new_tensor(
                 scale_factors).unsqueeze(1)
 
         # In mmdet 2.x, the class_id for background is num_classes.
         # i.e., the last column.
-        padding = batch_mlvl_scores.new_zeros(batch_size,
-                                              batch_mlvl_scores.shape[1], 1)
-        batch_mlvl_scores = torch.cat([batch_mlvl_scores, padding], dim=-1)
-
-        # Support exporting to onnx without nms
-        if with_nms and cfg.get('nms', None) is not None:
-            det_results = []
-            for (mlvl_bboxes, mlvl_scores,
-                 mlvl_conf_scores) in zip(batch_mlvl_bboxes, batch_mlvl_scores,
-                                          batch_mlvl_conf_scores):
-                # Filtering out all predictions with conf < conf_thr
-                conf_thr = cfg.get('conf_thr', -1)
-                if conf_thr > 0 and (not torch.onnx.is_in_onnx_export()):
-                    # TensorRT not support NonZero
-                    # add as_tuple=False for compatibility in Pytorch 1.6
-                    # flatten would create a Reshape op with constant values,
-                    # and raise RuntimeError when doing inference in ONNX
-                    # Runtime with a different input image (#4221).
-                    conf_inds = mlvl_conf_scores.ge(conf_thr).nonzero(
-                        as_tuple=False).squeeze(1)
-                    mlvl_bboxes = mlvl_bboxes[conf_inds, :]
-                    mlvl_scores = mlvl_scores[conf_inds, :]
-                    mlvl_conf_scores = mlvl_conf_scores[conf_inds]
-
-                det_bboxes, det_labels = multiclass_nms(
-                    mlvl_bboxes,
-                    mlvl_scores,
-                    cfg.score_thr,
-                    cfg.nms,
-                    cfg.max_per_img,
-                    score_factors=mlvl_conf_scores)
-                det_results.append(tuple([det_bboxes, det_labels]))
-
-        else:
-            det_results = [
-                tuple(mlvl_bs)
-                for mlvl_bs in zip(batch_mlvl_bboxes, batch_mlvl_scores,
-                                   batch_mlvl_conf_scores)
-            ]
+        padding = flatten_bboxes.new_zeros(num_imgs, flatten_bboxes.shape[1],
+                                           1)
+        flatten_cls_scores = torch.cat([flatten_cls_scores, padding], dim=-1)
+
+        det_results = []
+        for (bboxes, scores, objectness) in zip(flatten_bboxes,
```

Comment:
If this process is referenced from other repo, then citations and instructions need to be added.

Reply:
It seems that the order has been changed, and the overall logic has not changed?